components/editable-description.tsx

```diff
@@ -0,0 +1,93 @@
+"use client";
+
+import { useState } from "react";
+import { useRouter } from "next/navigation";
+
+import { updateProject } from "@/lib/actions";
+import { DesignProject } from "@/lib/types";
+
+import { CardContent, CardHeader, CardDescription } from "@/components/ui/card";
+import { Textarea } from "@/components/ui/textarea";
+import { toast } from "@/components/ui/use-toast";
+import { Button } from "./ui/button";
+
+export default function EditableDescription({
+  project,
+}: {
+  project: DesignProject;
+}) {
+  const router = useRouter();
+
+  const [description, setDescription] = useState(
+    project.description || "Untitled",
+  );
+  const [originalDescription, setOriginalDescription] = useState(
+    project.description || "Untitled",
+  );
+  const [editing, setEditing] = useState(false);
+
+  async function clickUpdateProject() {
+    try {
+      const updates = { description: description };
+      const result = await updateProject(project.id, updates);
+      setOriginalDescription(result?.description || description);
+      setEditing(false);
+      toast({ description: "Project updated succesfully." });
+      router.refresh();
+    } catch (error) {
+      toast({ description: "Project failed to be updated." });
+    }
+  }
+
+  function cancelEditing() {
+    setDescription(originalDescription); // Revert to original title
+    setEditing(false);
+  }
+
+  return (
+    <>
+      <CardHeader>
+        <div className="flex items-center gap-1">
+          Description
+          {!editing && (
+            <Button
+              variant="ghost"
+              className="p-2 text-muted-foreground"
+              onClick={() => setEditing(true)}
+            >
+              Edit
+            </Button>
+          )}
+        </div>
+      </CardHeader>
+      <CardContent>
+        <CardDescription className="whitespace-pre-line">
+          {!editing ? (
+            description
+          ) : (
+            <Textarea
+              defaultValue={description}
+              placeholder="Tell us a little bit about your project"
+              className="resize-y h-svh"
```

Comment:
ideally we'd want the height to match the content exactly, but it seems like shadcn/ui hasn't merged that functionality yet

https://github.com/shadcn-ui/ui/pull/1681

if you so desire you could copy paste this in, but definitely not necessary


I think svh is too high now :/ maybe h-72 is enough

Reply:
I have fixed this issue in different approach. If you like you can keep this.